modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java

```diff
@@ -7428,17 +7427,17 @@ private Map<Locale, String> _checkFriendlyURLMap(
 		Locale defaultLocale, Map<Locale, String> friendlyURLMap,
 		Map<Locale, String> titleMap) {
 
-		for (Map.Entry<Locale, String> friendlyURL :
-				friendlyURLMap.entrySet()) {
+		Map<Locale, String> newFriendlyURLMap = new HashMap<>(friendlyURLMap);
 
-			if (Validator.isNotNull(friendlyURL.getValue())) {
-				return friendlyURLMap;
-			}
+		if (!newFriendlyURLMap.containsKey(defaultLocale) ||
+			Validator.isNull(newFriendlyURLMap.get(defaultLocale))) {
+
+			newFriendlyURLMap.put(
+				defaultLocale,
+				_removeTrailingSlashes(titleMap.get(defaultLocale)));
 		}
 
-		return HashMapBuilder.put(
-			defaultLocale, _removeTrailingSlashes(titleMap.get(defaultLocale))
-		).build();
+		return newFriendlyURLMap;
```

Comment:
Why copy the map even in the case when you're not modifying it? That is wasteful. If there's nothing to do, return the map as is, otherwise use HashMapBuilder to copy and extend it.